### PR TITLE
Added Prim and Kruskal's Algorithm.

### PR DESCRIPTION
**Pull request by Tanmay2510**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by Tanmay2510** [comment]
Can you review the new pull request @HarshCasper @atarax665 ?

File: .gitignore [hunk 1]
```diff
@@ -154,3 +154,4 @@ venv.bak/
 dmypy.json
 .pyre/
 .pytype/
+
```
Comment (atarax665) [inline .gitignore]: This change is irrelevant.

File: Java/Algorithms/KruskalS_Algorithm.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	//@Override
+	public int compareTo(Edge o) {
+		//To Sort in increasing order
+		return this.weight - o.weight;  
+        	}
+      }
+public class KruskalS_Algorithm {
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+                int Vertices = scan.nextInt();
```
Comment (atarax665) [inline Java/Algorithms/KruskalS_Algorithm.java]: Inconsistent indentation.

File: Java/Algorithms/KruskalS_Algorithm.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			}
+		}
+	}
+       private static int findParent(int source, int[] parent) {
+		if(parent[source] == source) {
+			return source;
+		}
+		return findParent(parent[source],parent);
+	 }
+ }
```
Comment (atarax665) [inline Java/Algorithms/KruskalS_Algorithm.java]: Add sample Input/Output and space, time complexity in commented form.

File: Java/Algorithms/KruskalS_Algorithm.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+//It is an algroithm that help us to figure out the Minimum Spanning Tree of a given graph.
```
Comment (atarax665) [inline Java/Algorithms/KruskalS_Algorithm.java]: Enclose the description in multiline comment.

File: Java/Algorithms/Prim_Algorithm.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
+//It is an algorithm that help us to figure out  minimum spanning tree by selecting the adjacent vertices of already selected vertices.
```
Comment (atarax665) [inline Java/Algorithms/Prim_Algorithm.java]: Enclose the description in multiline comment.

File: Java/Algorithms/Prim_Algorithm.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+				int v1 = s.nextInt();
+				int v2 = s.nextInt();
+				int weight = s.nextInt();
+				adjacencyMatrix[v1][v2] = weight;
+				adjacencyMatrix[v2][v1] = weight;
+			}
+		
+			prims(adjacencyMatrix);
+       }
+}
```
Comment (atarax665) [inline Java/Algorithms/Prim_Algorithm.java]: Add sample Input/Output and space, time complexity in commented form.

File: Java/README.md [hunk 7]
```diff
@@ -10,12 +10,14 @@
 # Codes in the Java language
 
 ## Algorithms
-
+- [Kruskal's Algorithm](Algorithms/KruskalS_Algorithm.java)
+- [Prim Algorithm](Algorithms/Prim_Algorithm.java
```
Comment (atarax665) [inline Java/README.md]:
```suggestion
- [Prim Algorithm](Algorithms/Prim_Algorithm.java)
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Tanmay2510** [comment]
Please review it @HarshCasper @atarax665 .

File: Java/Algorithms/KruskalS_Algorithm.java (hunk 2 above)
Comment (HarshCasper) [reply Java/Algorithms/KruskalS_Algorithm.java]: Not resolved.

File: Java/Algorithms/KruskalS_Algorithm.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+		//To Sort in increasing order
+		return this.weight - o.weight;  
+        	}
+      }
+    public class KruskalS_Algorithm {
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+                int Vertices = scan.nextInt();
+		int Edges = scan.nextInt();
+                Edge input[] = new Edge[Edges];
```
Comment (HarshCasper) [inline Java/Algorithms/KruskalS_Algorithm.java]: Inconsistent Indentation

File: Java/README.md [hunk 9]
```diff
@@ -10,12 +10,14 @@
...
 ## Algorithms
-
+- [Kruskal's Algorithm](Algorithms/KruskalS_Algorithm.java)
+- [Prim Algorithm](Algorithms/Prim_Algorithm.java)
 - [Kadane's Algorithm](Algorithms/kadanes_Java.java)
 - [Russian Peasant Algorithm](Algorithms/RussianPeasant.java)
 - [Z-Algorithm](Algorithms/ZAlgorithm.java)
 - [The Boyer–Moore majority vote algorithm](Algorithms/Moore_Algorithm.java)
 
+
```
Comment (HarshCasper) [inline Java/README.md]: Revert this back

File: Java/Algorithms/Prim_Algorithm.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+			for(int i = 0; i < Edges; i++){
+				int v1 = s.nextInt();
+				int v2 = s.nextInt();
+				int weight = s.nextInt();
+				adjacencyMatrix[v1][v2] = weight;
+				adjacencyMatrix[v2][v1] = weight;
+			}
+		
+			prims(adjacencyMatrix);
+       }
```
Comment (HarshCasper) [inline Java/Algorithms/Prim_Algorithm.java]: Fix the indentation

File: Java/Algorithms/Prim_Algorithm.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+		}
+         	// Print edges of MST
+		for(int i = 1; i < v; i++){
+			if(parent[i] < i){
+				System.out.println(parent[i] + " "+ i +" "+ weight[i]);
+			}else{
+				System.out.println(i + " "+ parent[i] +" "+ weight[i]);
+			}
+		}
+	}
```
Comment (HarshCasper) [inline Java/Algorithms/Prim_Algorithm.java]: Format the Code

File: Java/Algorithms/KruskalS_Algorithm.java [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+		}
+	}
+       private static int findParent(int source, int[] parent) {
+		if(parent[source] == source) {
+			return source;
+		}
+		return findParent(parent[source],parent);
+	 }
+ }
+/*
```
Comment (HarshCasper) [inline Java/Algorithms/KruskalS_Algorithm.java]: Format the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
Please make a new PR or squash the commits before I merge this.

**Pull request closed without merging**